common/domain/replicationTaskExecutor.go

```diff
@@ -263,35 +288,71 @@ func (h *domainReplicationTaskExecutorImpl) handleDomainUpdateReplicationTask(ct
 		request.ConfigVersion = task.GetConfigVersion()
 	}
 
-	// todo (david.porter) reason through if this is compatible with the proposed merge strategy
-	// in active/active domains.
-	if existingDomain.FailoverVersion < task.GetFailoverVersion() {
+	if originalDomainState.FailoverVersion < task.GetFailoverVersion() {
 		recordUpdated = true
 		request.ReplicationConfig.ActiveClusterName = task.ReplicationConfig.GetActiveClusterName()
 		request.ReplicationConfig.ActiveClusters = task.ReplicationConfig.GetActiveClusters()
 		request.FailoverVersion = task.GetFailoverVersion()
 		request.FailoverNotificationVersion = notificationVersion
 		request.PreviousFailoverVersion = task.GetPreviousFailoverVersion()
-	} else if !existingDomain.ReplicationConfig.IsActiveActive() {
+	} else if !originalDomainState.ReplicationConfig.IsActiveActive() {
 		h.logger.Warn("the existing failover version was more recent, indicating that the domain replication message was out of date and is consequently being dropped",
-			tag.WorkflowDomainName(existingDomain.Info.Name),
-			tag.FailoverVersion(existingDomain.FailoverVersion),
+			tag.WorkflowDomainName(originalDomainState.Info.Name),
+			tag.FailoverVersion(originalDomainState.FailoverVersion),
 			tag.FailoverVersion(task.GetFailoverVersion()))
 	}
 
-	if existingDomain.ReplicationConfig.IsActiveActive() || task.ReplicationConfig.IsActiveActive() {
-		mergedActiveClusters, aaChanged := mergeActiveActiveScopes(existingDomain.ReplicationConfig.ActiveClusters, task.ReplicationConfig.ActiveClusters)
+	if intendedDomainState.ReplicationConfig.IsActiveActive() || task.ReplicationConfig.IsActiveActive() {
```

Comment:
I'm confused by the places where we use intendedDomainState and other places where we use originalDomainState